app/src/main/java/org/thoughtcrime/securesms/onboarding/pickname/PickDisplayNameActivity.kt

```diff
@@ -0,0 +1,156 @@
+import org.thoughtcrime.securesms.ui.outlinedTextFieldColors
+import org.thoughtcrime.securesms.util.setUpActionBarSessionLogo
+import javax.inject.Inject
+
+private const val EXTRA_PICK_NEW_NAME = "extra_pick_new_name"
+
+@AndroidEntryPoint
+class PickDisplayNameActivity : BaseActionBarActivity() {
+
+    @Inject
+    lateinit var viewModelFactory: PickDisplayNameViewModel.AssistedFactory
+
+    private val viewModel: PickDisplayNameViewModel by viewModels {
+        val pickNewName = intent.getBooleanExtra(EXTRA_PICK_NEW_NAME, false)
+        viewModelFactory.create(pickNewName)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setUpActionBarSessionLogo()
+
+        ComposeView(this)
+            .apply { setContent { DisplayNameScreen(viewModel) } }
+            .let(::setContentView)
+
+        lifecycleScope.launch {
+            viewModel.eventFlow.collect {
+                startMessageNotificationsActivity()
+            }
+        }
+    }
+
+    @Composable
+    private fun DisplayNameScreen(viewModel: PickDisplayNameViewModel) {
+        val state = viewModel.stateFlow.collectAsState()
+
+        AppTheme {
+            DisplayName(state.value, viewModel::onChange) { viewModel.onContinue(this) }
+        }
+    }
+
+    @Preview
+    @Composable
+    fun PreviewDisplayName() {
+        PreviewTheme(R.style.Classic_Dark) {
+            DisplayName(State())
+        }
+    }
+
+    @Composable
+    fun DisplayName(state: State, onChange: (String) -> Unit = {}, onContinue: () -> Unit = {}) {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(20.dp),
+            modifier = Modifier
+                .padding(horizontal = 50.dp)
+                .padding(bottom = 12.dp)
+        ) {
+            Spacer(modifier = Modifier.weight(1f))
+            Text(stringResource(state.title), style = MaterialTheme.typography.h4)
+            Text(
+                stringResource(state.description),
+                style = MaterialTheme.typography.base,
+                modifier = Modifier.padding(bottom = 12.dp))
+
+            OutlinedTextField(
+                value = state.displayName,
+                modifier = Modifier.contentDescription(R.string.AccessibilityId_enter_display_name),
+                onValueChange = { onChange(it) },
+                placeholder = { Text(stringResource(R.string.displayNameEnter)) },
+                colors = outlinedTextFieldColors(state.error != null),
+                singleLine = true,
+                keyboardActions = KeyboardActions(
+                    onDone = { onContinue() },
+                    onGo = { onContinue() },
+                    onSearch = { onContinue() },
+                    onSend = { onContinue() },
+                ),
+                isError = state.error != null,
+                shape = RoundedCornerShape(12.dp)
+            )
+
+            state.error?.let {
```

Comment:
The error isn't centered. Though I suspect this will go away once you change the above to your SessionOutlinedTextField